README.md

```diff
@@ -21,42 +21,139 @@
+Make sure that providers are compatible. Read more about providers [here](#rpc-providers)
 
-More API providers can be found [here](https://docs.starknet.io/documentation/tools/api-services/).
+Then run:
+```bash
+cargo run --release -- -c ./path/to/config.toml
+```
 
-*NOTE: we rely on [helios](https://github.com/a16z/helios) for both valid checkpoint values and consensus rpc urls*
+Once Beerus has started to verify that everything is working correctly, run this command:
+```
+hurl etc/rpc/starknet_getStateRoot.hurl
+```
+### Configuration
 
 | field   | example | description |
 | ----------- | ----------- | ----------- |
-| network | MAINNET or GOERLI | network to query |
-| eth_execution_rpc | https://eth-mainnet.g.alchemy.com/v2/YOURAPIKEY | untrusted l1 node provider url |
-| starknet_rpc | https://starknet-mainnet.g.alchemy.com/starknet/version/rpc/v0.6/YOURAPIKEY | untrusted l2 node provider url |
+| network | MAINNET or SEPOLIA| network to query |
+| eth_execution_rpc | https://eth-mainnet.g.alchemy.com/v2/{YOUR_API_KEY}| untrusted l1 node provider url |
+| starknet_rpc | https://starknet-mainnet.g.alchemy.com/starknet/version/rpc/v0.6/{YOUR_API_KEY}| untrusted l2 node provider url |
 | data_dir | tmp | `OPTIONAL` location to store both l1 and l2 data |
 | poll_secs | 5 | `OPTIONAL` seconds to wait for querying sn state |
 | rpc_addr | 127.0.0.1:3030 | `OPTIONAL` local address to listen for rpc reqs |
 | fee_token_addr | 0x049d36...e004dc7 | `OPTIONAL` fee token to check for `getBalance` |
 
-### Running Beerus for the first time
+When you select a network, check that `eth_execution_rpc` and `starknet_rpc` urls also point to their corresponding networks. For example:
 
-Copy the configuration file from `etc/conf/beerus.toml` and set up the API provider URLs in the copy.
+MAINNET
+```
+eth_execution_rpc = "https://eth-mainnet.g.alchemy.com/v2/{YOUR_API_KEY}"
+starknet_rpc = "https://starknet-mainnet.g.alchemy.com/starknet/version/rpc/v0.6/{YOUR_API_KEY}"
+```
+SEPOLIA
+```
+eth_execution_rpc = "https://eth-sepolia.g.alchemy.com/v2/{YOUR_API_KEY}"
+starknet_rpc = "https://starknet-sepolia.g.alchemy.com/starknet/version/rpc/v0.6/{YOUR_API_KEY}"
+```
 
-Then run:
+#### RPC providers
+Beerus relies on TWO untrusted RPC endpoints, one for L1 (Ethereum), and one for L2 (Starknet).
+As these are untrusted they will typically not be nodes run on your local host or your local network.
+
+##### Etherum RPC endpoint
+
+Etherum RPC provider must support the `eth_getProof` endpoint.
+
+You can check if the provider is compatible by running this command:
 ```bash
-cargo run --release -- -c ./path/to/config.toml
+# This is an example RPC url. Use your RPC provider url to check if the node is compatible.
+ETHERUM_RPC_URL="https://eth-sepolia.g.alchemy.com/v2/{YOUR_API_KEY}"
+curl --request POST \
+     --url $ETHERUM_RPC_URL \
+     --header 'content-type: application/json' \
+     --data '
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "eth_getProof",
```

Comment:
I don't think we actually use `eth_getProof` anywhere directly. It is also a part of the Ethereum RPC API, so it should be supported by all service providers.

Reply:
I'm pretty sure you're correct on both points, my bad.

Reply:
I'll remove that section then and just write a few words that we're using standard ethereum rpc api so all service providers will work.

Reply:
Just for reference here is a [link](https://github.com/a16z/helios?tab=readme-ov-file#supported-ethereum-execution-api-providers-) to helios readme about that endpoint. Maybe it used to be an issue, but is not anymore?

Reply:
As long as we're good with any Ethereum provider that Helios is good with, I believe the link above is enough.